homeassistant/helpers/redact.py

```diff
@@ -0,0 +1,72 @@
+"""Helpers to redact sensitive data."""
+from __future__ import annotations
+
+from collections.abc import Callable, Iterable, Mapping
+from typing import Any, TypeVar, cast, overload
+
+from homeassistant.core import callback
+
+REDACTED = "**REDACTED**"
+
+_T = TypeVar("_T")
+_ValueT = TypeVar("_ValueT")
+
+
+def partial_redact(
+    x: str | Any, unmasked_prefix: int = 4, unmasked_suffix: int = 4
+) -> str:
+    """Mask part of a string with *."""
+    if not isinstance(x, str):
+        return REDACTED
+
+    unmasked = unmasked_prefix + unmasked_suffix
+    if len(x) < unmasked * 2:
+        return REDACTED
+
+    suffix = x[-unmasked_suffix:] if unmasked_suffix else ""
+    return f"{x[:unmasked_prefix]}{'*' * (len(x) - unmasked)}{suffix}"
```

Comment:
```suggestion
    return f"{x[:unmasked_prefix]}***{suffix}"
```
We don't need to keep the length, 3 * seems sensible?

Reply:
Right, that makes sense 👍